discord-bot/main.py

```diff
@@ -52,8 +58,91 @@ async def github(interaction: discord.Interaction):
     await interaction.response.send_message(view=view, ephemeral=True)
 
 @bot.tree.command(description='Send a message to the channel.')
-async def message(interaction: discord.Interaction, content: str):
-    await interaction.response.send_message(content, ephemeral=True)
+async def confirm(interaction: discord.Interaction):
+    await interaction.response.send_message("Amazing", ephemeral=True)
+    if await user_has_confirmed(interaction, bot):
+        await interaction.followup.send("Confirmed", ephemeral=True)
+
+@bot.tree.command(description="Command me to send, change or delete my messages.")
+@app_commands.describe(action="Select the type of action you want to perform.")
+@app_commands.describe(target="Select the target ID to send, edit or delete.")
+@app_commands.describe(source="Copy or move from source message ID to target channel ID.")
+async def message(
+    interaction: discord.Interaction,
+    action: Literal["send", "edit", "delete", "copy", "move"],
+    target: Optional[str] = None,
+    source: Optional[str] = None):
+    if await user_has_access(interaction, ADMIN_ROLE_ID, minimum=True):
+        def check(m):
+            return m.author == interaction.user and m.channel == interaction.channel
+        await interaction.response.send_message(f"You selected to `{action}` a message", ephemeral=True)
+        if action == "send":
+            if source:
+                await interaction.followup.send("Source message ID not required for sending a message.", ephemeral=True)
+                return
+            if target is None:
+                await interaction.followup.send("Target channel ID not found. Message will be sent here instead.", ephemeral=True)
+                target = interaction.channel.id
+            try:
+                target_channel = bot.get_channel(int(target))
+            except ValueError:
+                await interaction.followup.send("Invalid channel ID.", ephemeral=True)
+                return
+            await interaction.followup.send("Please enter the content for the message:", ephemeral=True)
+            new_message = await bot.wait_for('message', check=check)
+            await target_channel.send(new_message.content)
+            await interaction.followup.send(f"Message sent to {target_channel.mention}", ephemeral=True)
+            await new_message.delete()
+
+        elif action in ['edit', 'delete']:
```

Comment:
The handling of both `edit` and `delete` is very nested and hard to understand.

Split up the two actions, even if it introduces some code duplication. 

```python
if action == "send":
   // handle send

if action == "edit":
   // handle edit

if action == "delete":
   // handle delete
```